prusti-encoder/src/encoders/type/domain.rs

```diff
@@ -0,0 +1,599 @@
+use prusti_rustc_interface::{
+    middle::ty::{self, TyKind, util::IntTypeExt},
+    abi,
+    span::symbol,
+};
+use task_encoder::{
+    TaskEncoder,
+    TaskEncoderDependencies,
+};
+use vir::{BinaryArity, UnaryArity, UnknownArity, FunctionIdent, CallableIdent, Arity, ToKnownArity};
+
+/// You probably never want to use this, use `SnapshotEnc` instead.
+/// Note: there should never be a dependency on `PredicateEnc` inside this
+/// encoder!
```

Comment:
Should it be public at all? Or something internal to the snapshot encoder?

Reply:
Made inaccessible with `pub(super)`